microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/core/ls/MicroProfileDelegateCommandHandler.java

```diff
@@ -117,12 +117,15 @@ private static MicroProfileProjectInfo getMicroProfileProjectInfo(List<Object> a
 			LOGGER.log(Level.WARNING, "Error while joining MicroProfile properties collector job", e);
 		}
 
-		Exception jobException = (Exception) job.getResult().getException();
+		Throwable jobException = job.getResult().getException();
 		if (jobException != null) {
-			if (jobException.getCause() != null) {
+			if (jobException.getCause() != null && jobException instanceof Exception) {
 				throw (Exception) jobException.getCause();
+			} else if (jobException.getMessage() != null) {
```

Comment:
why not 
```java
if ( jobException instanceof Exception) {
  throw jobException;
} else {
  throw new Exception(jobException);
}
```
?

Reply:
Yup, definitely better. It does make those 2 errors a lot bigger because they'll carry the original StackOverflow but that's probably what's wanted.